File: app/src/main/java/com/mithril/mobilegoldenleaf/models/Address.java
package com.mithril.mobilegoldenleaf.models;

import androidx.room.Entity;
import androidx.room.PrimaryKey;

@Entity
public class Address {
    @PrimaryKey(autoGenerate = true)
    private int id;
    private String street;
    private String zipCode;

    public Address() {

    }

    public Address(String street, String zipCode) {
        this.street = street;
        this.zipCode = zipCode;
    }

    public Address(int id, String street, String zipCode) {
        this.id = id;
        this.street = street;
        this.zipCode = zipCode;
    }


    public int getId() {
        return id;
    }

    public void setId(int id) {
        this.id = id;
    }

    public String getStreet() {
        return street;
    }

    public void setStreet(String street) {
        this.street = street;
    }

    public String getZipCode() {
        return zipCode;
    }

    public void setZipCode(String zipCode) {
        this.zipCode = zipCode;
    }

}

File: app/src/main/java/com/mithril/mobilegoldenleaf/models/Address.kt
package com.mithril.mobilegoldenleaf.models

import androidx.room.Entity
import androidx.room.Ignore
import androidx.room.PrimaryKey

@Entity
class Address {
    @PrimaryKey(autoGenerate = true)
    var id: Long = 0L
    var street: String? = null
    var zipCode: String? = null

    constructor()

    fun hasValidId(): Boolean {
        return id != 0L
    }

    @Ignore
    constructor(id: Long, street: String, zipCode: String) {
        this.id = id
        this.street = street
        this.zipCode = zipCode
    }

}
